eng/common/pipelines/templates/jobs/npm-publish.yml

```diff
@@ -0,0 +1,73 @@
+parameters:
+  Tag: 'latest'
+  ArtifactName: 'packages'
+  DependsOn: []
+  Environment: 'package-publish'
+  Registry: 'https://registry.npmjs.org/'
+
+jobs:
+- deployment: PublishPackage_${{ parameters.ArtifactName }}
+  displayName: 'Publish ${{ parameters.ArtifactName }} to ${{ parameters.Registry }}'  
+  environment: ${{ parameters.Environment }}
+  dependsOn: ${{ parameters.DependsOn }}
+  variables:
+    - name: ArtifactPath
+      value: $(Pipeline.Workspace)/${{ parameters.ArtifactName }}
+
+  templateContext:
+    type: releaseJob
+    isProduction: ${{ eq(parameters.Registry, 'https://registry.npmjs.org/') }}
+    inputs:
+    - input: pipelineArtifact
+      artifactName: ${{ parameters.ArtifactName }}
+      itemPattern: '**/*.tgz'
+      targetPath: $(ArtifactPath)
+
+  pool:
+    name: azsdk-pool
+    image: ubuntu-24.04
```

Comment:
Given this will run in a context that includes the image.yml template, do we want to use those vars here, or have this be entirely self contained?

Reply:
I want it to be self-contained because I expect a number of repos, think autorest.* repos, to consume this directly as a template instead of pulling in all the files from the repos. However, you do make a good point we might be able to consume it via path from the tools repo. I'll think about that one.